code_annotations/extensions/base.py

```diff
@@ -42,22 +42,28 @@ class SimpleRegexAnnotationExtension(AnnotationExtension, metaclass=ABCMeta):
     # Javascript and Python extensions for examples.
     lang_comment_definition = None
 
-    r"""
-    This format string/regex finds all comments in the file. The format tokens will be replaced with the
-    language-specific comment definitions defined in the sub-classes.
-
-    {multi_start} - start of the language-specific multi-line comment (ex. /*)
-    ([\d\D]*?)    - capture all of the characters...
-    {multi_end}   - until you find the end of the language-specific multi-line comment (ex. */)
-    |             - If you don't find any of those...
-    {single}      - start by finding the single-line comment token (ex. //)
-    (.*)          - and capture all characters until the end of the line
-
-    Returns a 2-tuple of:
-     - ("Comment text", None) in the case of a multi-line comment OR
-     - (None, "Comment text") in the case of a single-line comment
+    # This format string/regex finds all comments in the file. The format tokens will be replaced with the
+    # language-specific comment definitions defined in the sub-classes.
+    #
+    # Match groupdict will contain two named subgroups: 'comment' and 'prefixed_comment', of which at most
+    # one will be non-None.
+    comment_regex_fmt = r"""
+    {multi_start}           # start of the language-specific multi-line comment (ex. /*)
+    (?P<comment>  # Look for a multiline comment
+        [\d\D]*?            # capture all of the characters...
+    )
+    {multi_end}             # until you find the end of the language-specific multi-line
+                            # comment (ex. */)
+    |                       # If you don't find any of those...
+    (?P<prefixed_comment> # Look for a group of single-line comments
+        (?:                 # Non-capture mode
+            {single}        # start by finding the single-line comment token (ex. //)
+            .*              # and capture all characters until the end of the line
+            \n?             # followed by an optional carriage return
+            \ *             # and some empty space
+        )*                  # multiple times
+    )
     """
```

Comment:
Reminder to indent:
```suggestion
    comment_regex_fmt = r"""
        {multi_start}           # start of the language-specific multi-line comment (ex. /*)
        (?P<comment>  # Look for a multiline comment
            [\d\D]*?            # capture all of the characters...
        )
        {multi_end}             # until you find the end of the language-specific multi-line
                                # comment (ex. */)
        |                       # If you don't find any of those...
        (?P<prefixed_comment> # Look for a group of single-line comments
            (?:                 # Non-capture mode
                {single}        # start by finding the single-line comment token (ex. //)
                .*              # and capture all characters until the end of the line
                \n?             # followed by an optional carriage return
                \ *             # and some empty space
            )*                  # multiple times
        )
    """
```

Reply:
Right! I did not understand what you meant at first. Please check the updated indentation.